### PR TITLE
Spiders Buff and critters change

### PR DESCRIPTION
**Pull request by Jackrost**
Created new PR because merge issues

This PR buffing spiders and make them more big threat:
* damage and HP were buffed. They now deal little poison damage.
* No slipping!
* can now pull stuff (items and crited people)
* spiders can spawn webs that slowdown enemies and preys.

Another changes: 
* spiders now have separate event. You can tweak their count spawn.
* critters spawns now don't depend on chaos level (No more 20-30 mice spawn)

This PR conflict with another PR #13292

**Media**
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame

:cl: Jack Rost
- tweak:  Spiders have more damage and hp, cant slip and can pull stuff around now
- tweak: Critters spawn now doesn't depend on the chaos level
- add: Spiders can make web that slows down everyone but them now

### CODE REVIEW

File: Content.Server/Spider/SpiderSystem.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        }
+
+        private void OnSpawnNet(EntityUid uid, SpiderComponent component, SpiderWebActionEvent args)
+        {
+            if (args.Handled)
+                return;
+
+            var transform = Transform(uid);
+
+            if (!_mapManager.TryGetGrid(transform.GridUid, out var grid)) return;
```
Comment (AJCM-git) [inline Content.Server/Spider/SpiderSystem.cs:57]: put the return in a new line, not in the same one as the condition

File: Resources/Textures/Interface/Actions/meta.json [hunk 2]
```diff
@@ -54,6 +54,9 @@
         },
         {
             "name": "malfunction"
+        },
+        {
+            "name": "web"
```
Comment (AJCM-git) [inline Resources/Textures/Interface/Actions/meta.json:59]: where did you took this from? if its from the same place linked in the copyright in this file disregard this but if it isnt you need to link it

File: Content.Server/Spider/SpiderSystem.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        }
+
+        private bool IsTileBlockedByWeb(EntityCoordinates coords)
+        {
+            foreach (var entity in coords.GetEntitiesInTile())
+            {
+                if (HasComp<SpiderWebObjectComponent>(entity))
+                    return true;
+            }
+            return false;
```
Comment (AJCM-git) [inline Content.Server/Spider/SpiderSystem.cs:102]: i think these are too much special casing for something that should be a much more simpler check, it makes sense that spider webs can be spawned above walls, mobs, doors and pretty much anything except other webs since they aren't a hard object like a wall
Tldr is remove `IsTileBlocked()` check and keep `IsTileBlockedByWeb()`

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Resources/Textures/Interface/Actions/meta.json (hunk 2 above)
Comment (Jackrost) [reply Resources/Textures/Interface/Actions/meta.json:59]: It is mine

File: Content.Server/Spider/SpiderSystem.cs (hunk 3 above)
Comment (Jackrost) [reply Content.Server/Spider/SpiderSystem.cs:102]: Hm, spawn webs on walls? I.e. even through walls? Ok

Comment (Jackrost) [reply Content.Server/Spider/SpiderSystem.cs:102]: > Waste hours to test blocking checks
> Copy paste from mime
> Remove checks

File: Resources/Textures/Interface/Actions/meta.json (hunk 2 above)
Comment (AJCM-git) [reply Resources/Textures/Interface/Actions/meta.json:59]: Then if you agree with the `license` at the top of the file you should include something like 

> web made by jackfrost (github) for SS14"

in the `copyright` field

File: Content.Server/Spider/SpiderSystem.cs (hunk 3 above)
Comment (AJCM-git) [reply Content.Server/Spider/SpiderSystem.cs]: Oh i pointed at the mime cuz it used a GetEntitiesInTile and not the tileref thing you did before, mime does so many checks because its a wall and a hard object, it makes sense that a spider web can be on a wall

Comment (Jackrost) [reply Content.Server/Spider/SpiderSystem.cs]: I need be more specific about what you mean. For me it's not obvious that web can go through walls, because they are obviously solid

**Review by AJCM-git — CHANGES_REQUESTED** [review]

**Comment by mirrorcult** [comment]
please do not duplicate PRs just because you could not resolve git issues

**Comment by mirrorcult** [comment]
please do not duplicate PRs just because you could not resolve git issues

File: Content.Server/Spider/SpiderSystem.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+            // Spawn web in other directions
+            for (var i = 0; i < 4; i++)
+            {
+                var direction = (DirectionFlag) (1 << i);
+                coords = transform.Coordinates.Offset(direction.AsDir().ToVec());
+                    
+                if (!IsTileBlockedByWeb(coords))
+                    Spawn(component.WebPrototype, coords);
+            }
```
[the remark this answers is not in the document]
Comment (Jackrost) [reply Content.Server/Spider/SpiderSystem.cs:76]: Nope. It's opposite. I tested it few times already. And I tested this in online in battle situations
1) Blocking spread because center is blocked is lame. You can mispredict position for example and cooldown long enough to not let you easy spam it. And web is not so strong to not allow covering tiles around you. If you want cover missing tiles this blocking only makes your building worse. In narrow maintenance corridors you already limited with how much you can cover.
2) Why web check needs to be removed if this will spawn web on already created?

[the remark this answers is not in the document]
Comment (AJCM-git) [reply Content.Server/Spider/SpiderSystem.cs:76]: Okay, after testing this in-game i see what you were trying to do and this is valid. 
Do make a check so that when you fail to place webs, it doesnt use the action and a popup appears informing the user of this, something like "You can't place webs here! All cardinal directions already have webs!"

File: Content.Server/Spider/SpiderSystem.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        }
+
+        private void OnSpawnNet(EntityUid uid, SpiderComponent component, SpiderWebActionEvent args)
+        {
+            if (args.Handled)
+                return;
+
+            var transform = Transform(uid);
+
+            if (!_mapManager.TryGetGrid(transform.GridUid, out var grid))
```
Comment (AJCM-git) [inline Content.Server/Spider/SpiderSystem.cs:57]: you can just do a null check to see if the grid exists since the out var `grid` is not used elsewhere. Also add a popup explaining that "You can't place webs in space!" (Remember to remove MapManager dependency after this)

File: Content.Server/Spider/SpiderSystem.cs [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+                return;
+
+            args.Handled = true;
```
Comment (AJCM-git) [inline Content.Server/Spider/SpiderSystem.cs:60]: it changes nothing in this case since we dont return until the end after this, but we normally set the handled arguments true at the end of the method

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Server/Spider/SpiderSystem.cs (hunk 3 above)
Comment (metalgearsloth) [reply Content.Server/Spider/SpiderSystem.cs]: Please don't copy-paste code, especially not Rane's code. If you need wall spawning logic as an action it should be generic and not require us to double any bugfixes across several hundred lines of code in 2 systems.

Comment (AJCM-git) [reply Content.Server/Spider/SpiderSystem.cs]: Yeah the code in this review is old, it just does a single GenEntitiesInTile now so it should be fine

File: Content.Server/Spider/SpiderSystem.cs [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        [Dependency] private readonly IRobustRandom _robustRandom = default!;
+        [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<SpiderWebObjectComponent, ComponentStartup>(OnWebStartup);
+            SubscribeLocalEvent<SpiderComponent, ComponentStartup>(OnSpiderStartup);
+            SubscribeLocalEvent<SpiderComponent, SpiderWebActionEvent>(OnSpawnNet);
```
Comment (AJCM-git) [inline Content.Server/Spider/SpiderSystem.cs:35]: move the first two to a `SharedSpiderSystem`, the third one cant because it spawns something and spawning is not predicted so keep it here

File: Content.Server/Spider/SpiderWebObjectComponent.cs [hunk 8]
```diff
@@ -0,0 +1,7 @@
+namespace Content.Server.Spider
+{
+    [RegisterComponent]
+    public sealed class SpiderWebObjectComponent : Component
```
Comment (AJCM-git) [inline Content.Server/Spider/SpiderWebObjectComponent.cs:4]: move to  shared

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Server/Spider/SpiderSystem.cs [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+            if (transform.GridUid == null)
+            {
+                _popup.PopupEntity(Loc.GetString("spider-web-action-nogrid"), args.Performer, args.Performer);
+                return;
+            }
+
+            var coords = transform.Coordinates;
+
+            // TO-DO generic way to get certain coordinates
```
Comment (AJCM-git) [inline Content.Server/Spider/SpiderSystem.cs:35]: use TODO since to make this show up when someone searches that

File: Content.Shared/Spider/SpiderComponent.cs [hunk 10]
```diff
@@ -0,0 +1,22 @@
+using Content.Shared.Actions;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Shared.Spider
+{
+    [RegisterComponent, NetworkedComponent]
+    public sealed class SpiderComponent : Component
```
Comment (AJCM-git) [inline Content.Shared/Spider/SpiderComponent.cs:10]:
```suggestion
    [RegisterComponent, NetworkedComponent]
    [Access(typeof(SharedSpiderSystem))]
    public sealed class SpiderComponent : Component
```

File: Content.Shared/Spider/SpiderWebObjectComponent.cs [hunk 11]
```diff
@@ -0,0 +1,7 @@
+namespace Content.Shared.Spider
+{
+    [RegisterComponent]
+    public sealed class SpiderWebObjectComponent : Component
```
Comment (AJCM-git) [inline Content.Shared/Spider/SpiderWebObjectComponent.cs:4]:
```suggestion
    [RegisterComponent, NetworkedComponent]
    [Access(typeof(SharedSpiderSystem))]
    public sealed class SpiderWebObjectComponent : Component
```

File: Content.Server/Spider/SpiderSystem.cs [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+{
+    public sealed class SpiderSystem : EntitySystem
+    {
+        [Dependency] private readonly PopupSystem _popup = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<SpiderComponent, SpiderWebActionEvent>(OnSpawnNet);
+        }
```
Comment (AJCM-git) [inline Content.Server/Spider/SpiderSystem.cs:18]: Inherit `SharedSpiderSystem`  
Im pretty sure you can just subscribe the `SpiderWebActionEvent` there then handle it here, tho i dont remember if actions had some quirkiness with that so disregard this last bit if it doesn't work

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Also reduce the scope of your cs files, for example

`namespace Content.Server.Spider;`

instead of 
```
namespace Content.Server.Spider 
{ 
	... 
}
```

**Review by AJCM-git — APPROVED** [review]
